pkg/volume/cinder/attacher.go

```diff
@@ -278,6 +314,15 @@ func (detacher *cinderDiskDetacher) Detach(deviceMountPath string, nodeName type
 		instanceID = instanceID[(ind + 1):]
 	}
 
+	pending, volumeStatus, err := detacher.cinderProvider.OperationPending(volumeID)
+	if err != nil {
+		return err
+	}
+	if pending {
+		glog.Infof("status of volume %q is %s, wait it to be finished", volumeID, volumeStatus)
+		return nil
```

Comment:
If you return `nil` here then it means that the volume is detached, right? You should probably wait here for the pending operation to complete (with the same exponential backoff as above?)